core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -787,9 +787,14 @@ private void copyLogSegment(UnifiedLog log, LogSegment segment, long nextSegment
             long bytesLag = log.onlyLocalLogSegmentsSize() - log.activeSegment().size();
             String topic = topicIdPartition.topic();
             int partition = topicIdPartition.partition();
-            brokerTopicStats.topicStats(topic).recordRemoteCopyLagBytes(partition, bytesLag);
+
+            BrokerTopicMetrics brokerTopicMetrics = brokerTopicStats.topicStats(topic);
+            brokerTopicMetrics.recordRemoteCopyLagBytes(partition, bytesLag);
+            brokerTopicStats.allTopicsStats().recordRemoteCopyLagBytes(topic, brokerTopicMetrics.remoteCopyLagBytes());
```

Comment:
nit

perhaps worth adding a comment on why we are using `brokerTopicMetrics.remoteCopyLagBytes()` instead simply updating with `bytesLag` here. I think the answer is that we want to update the "topic" level metric with an aggregate value across partitions and `bytesLag` just represents this partition.